From a79190b28a94c1135f0e6ceb2e49865e90b50f6b Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Wed, 17 Jul 2024 18:42:14 +0200
Subject: [PATCH] [Chronik] Fix: Use actual CBlockHeader in
 GetFirstBlockTxOffset (#28)

Currently, `GetFirstBlockTxOffset` results in wrong values for AuxPow (which are longer than 80 bytes), resulting in bad data_pos values.

Original-Commit: 42f55b94abade40b957182850c37a12af1ed2bfd
Adaptation: Preserve current block_data_pos API; serialize the actual AuxPoW header.

diff --git a/chronik/chronik-cpp/chronik_bridge.cpp b/chronik/chronik-cpp/chronik_bridge.cpp
index bae29312a1f5d407d4c578de8fe68eadd756a3ae..6fd9f2430aa74225645be8d6042a416ef21a8c57 100644
--- a/chronik/chronik-cpp/chronik_bridge.cpp
+++ b/chronik/chronik-cpp/chronik_bridge.cpp
@@ -111,7 +111,7 @@ chronik_bridge::BlockTx BridgeBlockTx(bool isCoinbase, const CTransaction &tx,
 
 inline size_t GetFirstBlockTxOffset(const CBlock &block,
                                     const size_t block_data_pos) {
-    return block_data_pos + ::GetSerializeSize(CBlockHeader()) +
+    return block_data_pos + ::GetSerializeSize(block.GetBlockHeader()) +
            GetSizeOfCompactSize(block.vtx.size());
 }
 
